core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -15,13 +15,18 @@
 void imx_configure_tzasc(void)
 {
 
-	int i;
+	int i = 0;
+	int end = 1;
 	vaddr_t addr[2] = {0};
 
 	addr[0] = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
-	addr[1] = core_mmu_get_va(TZASC2_BASE, MEM_AREA_IO_SEC);
 
-	for (i = 0; i < 2; i++) {
+	if (!(soc_is_imx6ul() || soc_is_imx6ull())) {
```

Comment:
would it be possible to check the definition of TZASC2_BASE instead?
like #if defined (TZASC2_BASE)

Reply:
In theory this is possible, however I am not a fan of defines in the C file. Are there more i.MX6 processors which only support one MMDC which are supported by OP-TEE?

Reply:
i think that only 6q and 6qp have do have 2 controllers.
i am also not a fan of defines in C file but i rather have something generic that can take a configuration from conf.mk file rather having to modify couple of files each time i am adding a new soc to support.
same thing goes with the addition of drivers

Reply:
Actually adding more defines will make it harder for long term maintain. whether soc_is_imx6ul, or similar logic is_enabled(CFG_MX6UL), both good, but need introduce some logic in OP-TEE for is_enabled(CFG_xx)

Reply:
+1 for introducing `is_enabled(CFG_xx)` if you think it makes sense here. Such a construct is good for other reasons: improving compile test coverage, and making it harder to make drastic changes in the code logic (which would make the code more difficult to read, maintain etc.).

Reply:
I added a naive `IS_CONFIGURED` macro, please take a look.